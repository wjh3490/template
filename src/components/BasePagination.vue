<template>
  <div class="pagination-container">
    <el-pagination
    class="pagination"
      :background="background"
      :current-page.sync="currentPage"
      :page-size.sync="pageSize"
      :layout="layout"
      :page-sizes="pageSizes"
      :total="total"
      @size-change="handleSizeChange"
      @current-change="handleCurrentChange"
    />
  </div>
</template>

<script>
export default {
  name: 'Pagination',
  props: {
    total: {
      required: true,
      type: Number
    },
    page: {
      type: Number,
      default: 1
    },
    limit: {
      type: Number,
      default: 20
    },
    pageSizes: {
      type: Array,
      default() {
        return [10, 20, 30, 40];
      }
    },
    layout: {
      type: String,
      default: 'total, sizes, prev, pager, next, jumper'
    },
    background: {
      type: Boolean,
      default: true
    }
  },
  computed: {
    currentPage: {
      get() {
        return this.page;
      },
      set(val) {
        this.$emit('update:page', val);
      }
    },
    pageSize: {
      get() {
        // 3
        return this.limit;
      },
      set(val) {
        // 1
        this.$emit('update:limit', val);
      }
    }
  },

  methods: {
    handleSizeChange(val) {
      // 2
      this.$emit('pagination', { page: this.currentPage, limit: val });
    },
    handleCurrentChange(val) {
      this.$emit('pagination', { page: val, limit: this.pageSize });
    }
  }
};
</script>

<style scoped></style>
