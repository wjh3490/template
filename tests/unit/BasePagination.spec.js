import { shallowMount } from '@vue/test-utils'
import BasePagination from '../../src/components/BasePagination.vue'
describe('BasePagination.vue', () => {
    it('toggle click', () => {
        const wrapper = shallowMount(BasePagination, {
            propsData: {
                total: 10
            }
        }
        )
        const mockFn = jest.fn();
    wrapper.setMethods({
      'handleCurrentChange': mockFn
    });
        console.log( wrapper.find('.pagination'))
    wrapper.find('.pagination').vm.$emit('current-change');
    expect(mockFn).toHaveBeenCalledTimes(1);

    })
  
  })