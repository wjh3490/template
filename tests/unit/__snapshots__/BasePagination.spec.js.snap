// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasePagination.vue toggle click 1`] = `
<div class="pagination-container">
  <el-pagination background="true" current-page="1" page-size="20" layout="total, sizes, prev, pager, next, jumper" page-sizes="10,20,30,40" total="10"></el-pagination>
</div>
`;
